linkerd/app/core/src/control.rs

```diff
@@ -87,12 +93,14 @@ impl Config {
             // Ensure individual endpoints are driven to readiness so that the balancer need not
             // drive them all directly.
             .push_on_service(svc::layer::mk(svc::SpawnReady::new))
+            .instrument(|t: &self::client::Target| tracing::info_span!("endpoint", addr = %t.addr))
             .push(self::resolve::layer(dns, resolve_backoff))
             .push_on_service(self::control::balance::layer())
             .into_new_service()
             .push(metrics.to_layer::<classify::Response, _, _>())
             .push(self::add_origin::layer())
             .push_on_service(svc::layers().push_spawn_buffer(self.buffer_capacity))
+            .instrument(|c: &ControlAddr| tracing::info_span!("controller", addr = %c.addr))
```

Comment:
just double-checking that there are, in fact, `INFO`-level events (or `WARN`/`ERROR`) that will occur inside these, right?

Reply:
yeah we mostly want connection errors, etc, to be logged with context.